devbin/rotate_keys.py

```diff
@@ -87,15 +109,73 @@ def add_new_key(self, k: IAMKey):
         self.keys.insert(0, k)
 
     def list_keys(self) -> str:
```

Comment:
another not particularly important performance note: I'd probably write a function like this to take a `io.TextIOBase` and return None. You can pass StringIO if you want a string back, but you could also just pass the stream of interest.

Reply:
I'll just make this change it feels cleaner.